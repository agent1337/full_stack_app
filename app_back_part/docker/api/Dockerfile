FROM node:lts-jessie

WORKDIR /home/node/app
COPY /app/package*.json /home/node/app/
COPY /app/tsconfig.json /home/node/app/
RUN npm install
CMD npm run dev
EXPOSE 9000
