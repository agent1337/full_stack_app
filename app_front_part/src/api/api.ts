import axios from 'axios'


export interface User {
    id: string,
    email: string
}

class Api {
    // @ts-ignore
    private api

    constructor() {
        this.api = axios.create({
            baseURL: 'http://localhost:3000'
        });
    }

    public async getUsers(): Promise<User[]> {
        // console.log()
        return await this.api.get("/users/getAllUsers")
    }
}

export default new Api()