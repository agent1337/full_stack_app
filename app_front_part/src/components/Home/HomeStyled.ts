import {makeStyles} from "@material-ui/core/styles";

export const useStyles = makeStyles({
    wrapper: {
        width: '45%',
        display: 'flex',
        justifyContent: 'space-evenly',
        alignItems: 'center'
    }
});
