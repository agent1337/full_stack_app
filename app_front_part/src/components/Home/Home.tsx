import history from "../../history";
import React, {useEffect} from 'react'
import {Button} from '@material-ui/core'
import {useStyles} from './HomeStyled'
import api from "../../api/api";


const Home = () => {
    const classes = useStyles()

    // test
    useEffect(() => {
        console.log('rendered');
        const fetchUsers = async () => {
            const users = await api.getUsers();
            // @ts-ignore
            return users.data
        };
        fetchUsers()
            .then((res) => {
                console.log(res)
            })

    }, []);


    return (
        <div className={classes.wrapper}>

            <Button
                data-testid="button-login"
                onClick={() => {
                    history.push('/users/login')
                }}
                variant="contained"
                color="secondary"
            >
                Login
            </Button>

            <Button
                data-testid="button-signUp"
                onClick={() => {
                history.push('/users/signUp')
            }}
                variant="contained"
                color="secondary"
            >
                Sign Up
            </Button>

        </div>
    )
}

export default Home