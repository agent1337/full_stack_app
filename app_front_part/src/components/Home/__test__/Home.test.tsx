import React from 'react'
import {render,fireEvent} from '@testing-library/react'
import Home from '../Home'
import api, {User} from "../../../api/api";
import {unmountComponentAtNode} from "react-dom";

let container: any = null;

describe('Home', function () {
    beforeEach(() => {
        // подготавливаем DOM-элемент, куда будем рендерить
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        // подчищаем после завершения
        unmountComponentAtNode(container);
        container.remove();
        container = null;
    });

    describe('Home', function () {

        it('should to match snapshot', function () {
            const {container} = render(<Home/>, {});

            expect(container).toMatchSnapshot()
        });


        it('should send get req on mount', function () {
            const axiosGetSpy = jest.spyOn(api, 'getUsers').mockResolvedValue([{} as User]);

            render(<Home/>, {})

            expect(axiosGetSpy).toBeCalledWith()
        });


        it('user clicks button login', function () {
            const { getByTestId } = render(<Home/>, {});

            fireEvent.click(getByTestId('button-login'))
        });
    });
});