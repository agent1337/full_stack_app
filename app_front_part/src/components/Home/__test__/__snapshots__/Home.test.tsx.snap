// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home should to match snapshot 1`] = `
<div>
  <div
    class="makeStyles-wrapper-1"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
      data-testid="button-login"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Login
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
      data-testid="button-signUp"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Sign Up
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
